modules/new_serverpod_auth/serverpod_auth_core/serverpod_auth_core_server/lib/src/session/business/auth_sessions_token_manager.dart

```diff
@@ -0,0 +1,140 @@
+import 'package:serverpod/protocol.dart';
+import 'package:serverpod/serverpod.dart';
+
+import '../../common/business/token_manager.dart';
+import '../../generated/protocol.dart';
+import 'auth_session_config.dart';
+import 'auth_sessions.dart';
+
+/// Token manager adapter for [AuthSessions].
+///
+/// This class is used to bridge the gap between the [AuthSessions] and the [TokenManager] interface.
+/// It delegates all operations to the [AuthSessions] instance.
+class AuthSessionsTokenManager implements TokenManager {
+  /// The name of the token issuer.
+  static String get tokenIssuerName => AuthStrategy.session.name;
+
+  /// The [AuthSessions] instance.
+  final AuthSessions _authSessions;
+
+  /// Creates a new [AuthSessionsTokenManager] instance.
+  AuthSessionsTokenManager({required final AuthSessionConfig config})
+      : _authSessions = AuthSessions(config: config);
+
+  @override
+  Future<AuthSuccess> issueToken(
+    final Session session, {
+    required final UuidValue authUserId,
+    required final String method,
+    final Set<Scope>? scopes,
+    final Transaction? transaction,
+  }) {
+    return _authSessions.createSession(
+      session,
+      authUserId: authUserId,
+      method: method,
+      scopes: scopes,
+      transaction: transaction,
+    );
+  }
+
+  @override
+  Future<List<TokenInfo>> listTokens(
+    final Session session, {
+    required final UuidValue? authUserId,
+    final String? method,
+    final String? tokenIssuer,
+    final Transaction? transaction,
+  }) async {
+    if (tokenIssuer != null && tokenIssuer != tokenIssuerName) {
+      return [];
+    }
+
+    return (await _authSessions.admin.findSessions(
+      session,
+      authUserId: authUserId,
+      method: method,
+      transaction: transaction,
+    ))
+        .map((final element) => TokenInfo(
+              userId: element.authUserId.toString(),
+              tokenId: element.id.toString(),
+              tokenIssuer: tokenIssuerName,
+              scopes: element.scopeNames.map(Scope.new).toSet(),
+              method: element.method,
+            ))
+        .toList();
+  }
+
+  @override
+  Future<void> revokeAllTokens(
+    final Session session, {
+    required final UuidValue? authUserId,
+    final Transaction? transaction,
+    final String? method,
+    final String? tokenIssuer,
+  }) async {
+    if (tokenIssuer != null && tokenIssuer != tokenIssuerName) return;
+
+    final deletedSessions = await _authSessions.admin.deleteSessions(
+      session,
+      transaction: transaction,
+      authUserId: authUserId,
+      method: method,
+    );
+
+    for (final (authUserId, sessionId) in deletedSessions) {
```

Comment:
With properly names record this becomes: `(:authUserId, :sessionId)` so not cumbersome at all, and robust against renaming, re-ordering, etc.